JamfPro/JamfPro/JamfPro.pq

```diff
@@ -271,751 +117,484 @@ UAPIResource = (baseurl as text, relativepath as text, optional query as record,
-    #"Expanded computerGroupDetails.computergroup" = Table.ExpandListColumn(#"Expanded computerGroupDetails", "computerGroupDetails.computergroup"),
-    #"Expanded computerGroupDetails.computergroup1" = Table.ExpandRecordColumn(#"Expanded computerGroupDetails.computergroup", "computerGroupDetails.computergroup", {"id", "name", "mac_address", "alt_mac_address", "serial_number"}, {"computer_id", "computer_name", "computer_mac_address", "computer_alt_mac_address", "computer_serial_number"}),
-    #"Removed Columns" = Table.RemoveColumns(#"Expanded computerGroupDetails.computergroup1",{"computer_mac_address", "computer_alt_mac_address", "computer_serial_number"})
-    //#"Renamed Columns" = Table.RenameColumns(#"Removed Columns",{{"computerGroupDetails.computergroup.id", "computer_id"}, {"computerGroupDetails.computergroup.name", "computer_name"}})
-in
-    #"Removed Columns";
 
+        temp_table = Table.ExpandRecordColumn(deviceGroups, "mobileDeviceGroupDetails", {"mdgroup"}, {"mobileDeviceGroupDetails.mdgroup"}),
+        expanded_table = Table.ExpandListColumn(temp_table, "mobileDeviceGroupDetails.mdgroup"),
+        expanded_table_2 = Table.ExpandRecordColumn(expanded_table, "mobileDeviceGroupDetails.mdgroup", {"id", "name", "mac_address", "udid", "wifi_mac_address", "serial_number"}, {"device_id", "device_name", "device_mac_address", "device_udid", "device_wifi_mac_address", "device_serial_number"})
+    in
+        expanded_table_2;
 
+getMobileDeviceGroupDetails = (id as number, baseurl as text) as any =>
+    let 
+        jsonMDDetails = getJssResource(id,baseurl,"/mobiledevices/id/"),
+        mobile_device = jsonMDDetails[mobile_device],
+        general = mobile_device[general],
+        location = mobile_device[location], 
+        purchasing = mobile_device[purchasing],
+        security = mobile_device[security],
+        network = mobile_device[network],
+        extensionAttributes = mobile_device[extension_attributes],
+        extAttrRec = [extAttrs=extensionAttributes],
+        applications = mobile_device[applications],
+        applicationsRec = [mdApps=applications],
+        result = Record.Combine({general,location,purchasing,security,network,extAttrRec,applicationsRec})
+    in
+        result;
+
+getJssResource = (id as number, baseurl as text,path as text) as any =>
+    let
+        response = JSSResource(baseurl,path & Number.ToText(id)),
+        result = Json.Document(response)
+    in
+        result;
+ 
 // Data Source Kind description
 JamfPro = [
     TestConnection = (dataSourcePath) => {"JamfPro.Contents", dataSourcePath},
     Authentication = [
-        
         UsernamePassword = [
-             UsernameLabel="Enter your Jamf Instance User name: ",
-    PasswordLabel="Enter your Jamf instance password: "
+            UsernameLabel="Enter your Jamf Instance User name: ",

```

Comment:
Let's change `Jamf Instance User name` to `Jamf Pro username`. Same for the password label below.